### ISSUE TITLE
Maven compatibility issues 

### ISSUE THREAD

**Issue by syphr42** [openhab/static-code-analysis#45]
The ESH code formatter sets the max line width to 120 while the LineLengthCheck configured in this tool for CheckStyle flags lines over 80 characters as priority 1.

* What is the desired max line length for ESH & openHAB?
* The formatter and the CheckStyle rule should be aligned.

**Comment by kaikreuzer** [comment]
Thanks for noticing! The desired length is indeed 120.

**Comment by svilenvul** [comment]
I am little bit surprised, as the LineLengthCheck is not included in the [checkstyle ruleset](https://github.com/openhab/static-code-analysis/blob/master/src/main/resources/rulesets/checkstyle/rules.xml). Maybe some FindBugs or PMD check is logging this, but I haven't seen a failing build.

Can you provide an example in which bundle and file have you received this warning ?

**Comment by syphr42** [comment]
It looks like there is a different problem. I was trying to put together a PR to fix the line length and I realized what @svilenvul mentioned - line length is not one of the checks in the rules.xml.

So digging into this some more, I found that the rules.xml is not being used at all. Instead, Checkstyle is executing sun_checks.xml (appears to be the default). It looks like the settings from the plugin are not executing, but the the goals are - it runs Checkstyle, FindBugs, and PMD. 

Because the properties are not being used, it is dropping the checkstyle-result.xml in target/ rather than target/code-analysis/ so the merge fails and report.html is zero bytes. If I manually copy the checkstyle-result.xml to the code-analysis folder and run again without 'clean' it will generate the report, but it is of course using the default rules.

Originally I was executing as
```
mvn clean install -P check
```

But in response to @kaikreuzer responding to my forum post [here](https://community.openhab.org/t/static-code-analysis-tool/24489/45) I just did a normal

```
mvn clean install
```

and I got the same result.

I then checked out master from the openhab2-addons repo and ran 'mvn clean install' on the Chromecast binding with the same result:

```
$ mvn clean install
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO] Fetching p2.index from http://download.eclipse.org/smarthome/updates-stable/
[INFO] Adding repository http://download.eclipse.org/smarthome/updates-stable
[INFO] Fetching p2.index from https://dl.bintray.com/openhab/p2/openhab-core/2.1.x/ (172B)
[INFO] Fetching p2.index from https://dl.bintray.com/openhab/p2/openhab-core/2.1.x/ (172B)
[INFO] Adding repository https://dl.bintray.com/openhab/p2/openhab-core/2.1.x
[INFO] Adding repository https://dl.bintray.com/openhab/p2/openhab-deps-repo/1.0.10
[INFO] Resolving dependencies of MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO] Resolving class path of MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Chromecast Binding 2.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ org.openhab.binding.chromecast ---
[INFO] Deleting /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:build-qualifier (default-build-qualifier) @ org.openhab.binding.chromecast ---
[INFO] The project's OSGi version is 2.1.0.201704040041
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-id (default-validate-id) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-version (default-validate-version) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:resources (default-resources) @ org.openhab.binding.chromecast ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/src/main/resources
[INFO] 
[INFO] --- tycho-compiler-plugin:1.0.0:compile (default-compile) @ org.openhab.binding.chromecast ---
[INFO] Compiling 4 source files to /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/classes
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:compile (default) @ org.openhab.binding.chromecast ---
[INFO] Changes detected - recompiling the module!
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:testResources (default-testResources) @ org.openhab.binding.chromecast ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/src/test/resources
[INFO] 
[INFO] --- target-platform-configuration:1.0.0:target-platform (default-target-platform) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:package-plugin (default-package-plugin) @ org.openhab.binding.chromecast ---
[INFO] Building jar: /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- tycho-p2-plugin:1.0.0:p2-metadata-default (default-p2-metadata-default) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- static-code-analysis:0.0.3:checkstyle (default) @ org.openhab.binding.chromecast ---
[INFO] Adding dependency to checkstyle:7.2
[INFO] There are 161 errors reported by Checkstyle 7.2 with sun_checks.xml ruleset.
[WARNING] Unable to locate Source XRef to link to - DISABLED
[INFO] 
[INFO] --- static-code-analysis:0.0.3:pmd (default) @ org.openhab.binding.chromecast ---
[WARNING] Unable to locate Source XRef to link to - DISABLED
[WARNING] Unable to locate Source XRef to link to - DISABLED
[INFO] 
[INFO] --- static-code-analysis:0.0.3:findbugs (default) @ org.openhab.binding.chromecast ---
[INFO] Adding dependency to findbugs:3.0.1
[INFO] Adding dependency to bug-pattern:1.2.4
[INFO] Fork Value is true
     [java] Warnings generated: 2
[INFO] Done FindBugs Analysis....
[INFO] 
[INFO] --- static-code-analysis:0.0.3:report (default) @ org.openhab.binding.chromecast ---
Error 
  java.io.FileNotFoundException:
  /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/checkstyle-result.xml (No such file or directory)
[ERROR] java.io.FileNotFoundException: /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/checkstyle-result.xml (No such file or directory)
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
[Fatal Error] :-1:-1: Premature end of file.
[WARNING] Cann't select /sca/file/message nodes from /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/f48f91d0_SecondMerge.xml. Empty NodeList will be returned.
org.xml.sax.SAXParseException; Premature end of file.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:205)
	at org.openhab.tools.analysis.report.ReportUtility.selectNodes(ReportUtility.java:346)
	at org.openhab.tools.analysis.report.ReportUtility.appendToSummary(ReportUtility.java:297)
	at org.openhab.tools.analysis.report.ReportUtility.execute(ReportUtility.java:194)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:153)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
[INFO] Empty report will not be appended to the summary report.
[Fatal Error] :-1:-1: Premature end of file.
[WARNING] Cann't select /sca/file/message[@priority=1] nodes from /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/f48f91d0_SecondMerge.xml. Empty NodeList will be returned.
org.xml.sax.SAXParseException; Premature end of file.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:205)
	at org.openhab.tools.analysis.report.ReportUtility.selectNodes(ReportUtility.java:346)
	at org.openhab.tools.analysis.report.ReportUtility.checkForErrors(ReportUtility.java:262)
	at org.openhab.tools.analysis.report.ReportUtility.execute(ReportUtility.java:199)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:153)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
[INFO] 
[INFO] --- maven-install-plugin:2.3.1:install (default-install) @ org.openhab.binding.chromecast ---
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.pom
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/p2content.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT-p2metadata.xml
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/p2artifacts.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT-p2artifacts.xml
[INFO] 
[INFO] --- tycho-p2-plugin:1.0.0:update-local-index (default-update-local-index) @ org.openhab.binding.chromecast ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 17.319s
[INFO] Finished at: Mon Apr 03 20:41:51 EDT 2017
[INFO] Final Memory: 56M/524M
[INFO] ------------------------------------------------------------------------
```
In particular, notice this line:
```
[INFO] There are 161 errors reported by Checkstyle 7.2 with sun_checks.xml ruleset.
```

**Comment by svilenvul** [comment]
Which version of Maven is installed ?

He had recently similar issues with 3.0.5 and even 3.2.5. - [see](https://github.com/openhab/static-code-analysis/issues/47#issuecomment-291098838)

**Comment by syphr42** [comment]
I was on Maven 3.1.0. Upgrading to 3.3.9 solved the problem.

**Comment by martinvw** [comment]
@kaikreuzer @svilenvul Is this a requirment we want to add to the information regarding building the project, or should we invest more time in finding the cause of this?

**Comment by svilenvul** [comment]
I think its worth to invest some time and make it Maven 3.x compatible.

I have reproduced it locally and I have tried to find the reason, but without a success until now.

@martinvw, I will appreciate your help.

**Comment by kaikreuzer** [comment]
@svilenvul & @martinvw Note that I am currently having HUGE issues wrt Maven versions of our build, see https://community.openhab.org/t/jenkins-build-problems/25977.
I still haven't figured out why all of this happens, but it somehow correlated to doing and using the static analysis tool 0.0.3 release. Might this plugin introduce dependencies that are incompatible with dependencies of the Karaf tooling?

**Comment by martinvw** [comment]
I'm comparing the dependency tree between maven 3.2 and 3.3 to check whether something stands out there.

https://maven.apache.org/plugins/maven-dependency-plugin/examples/resolving-conflicts-using-the-dependency-tree.html

**Comment by martinvw** [comment]
I think this line is might be a main suspect for maven 3.3 compatibility:

https://github.com/openhab/static-code-analysis/blob/master/pom.xml#L23

**Comment by martinvw** [comment]
Only changing that, does not fix it, I'll try to find some more time later today.

**Comment by martinvw** [comment]
@svilenvul As you might have seen I tried some things, I might have found a way/workaround, by populating the configuration straight away. https://github.com/martinvw/static-code-analysis/blob/feature/maven-compatibility/src/main/java/org/openhab/tools/analysis/tools/CheckstyleChecker.java#L111

But I was curious why it has been constructed to extend the list of userProperties, from the API documentation I could not see whether this was a recommended and supported way.

**Comment by svilenvul** [comment]
> But I was curious why it has been constructed to extend the list of userProperties, from the API documentation I could not see whether this was a recommended and supported way.

Thanks for pointing me in the right direction !

Actually I have found something that seems to be working - https://web.archive.org/web/20150509231102/http://docs.codehaus.org/display/MAVENUSER/Mojo+Developer+Cookbook#MojoDeveloperCookbook-StoringProperties

**Comment by martinvw** [comment]
So this works in both new and old versions of maven, great!

**Comment by svilenvul** [comment]
I have tried it with 3.0.5 and 3.3.9, and it seems to work.

**Comment by martinvw** [comment]
Great, I also fixed some other (potential) issues while working on it, I might do a PR for that later on.

**Comment by martinvw** [comment]
I just tested it locally, it works perfect!

**Comment by svilenvul** [comment]
>Great, I also fixed some other (potential) issues while working on it, I might do a PR for that later on. 

Great, thanks !
